### PR TITLE
plan: check invalid task after physical optimization

### PR DESCRIPTION
**Pull request by zz-jason**
To avoid the panic caused by "nil pointer dereference"

### CODE REVIEW

**Comment by zz-jason** [comment]
@lamxTyler @winoros @shenli PTAL

File: plan/optimizer.go [hunk 1]
```diff
@@ -145,17 +145,26 @@ func logicalOptimize(flag uint64, logic LogicalPlan) (LogicalPlan, error) {
...
+
+	t, err := logic.findBestTask(prop)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	p := t.plan()
-	p.ResolveIndices()
-	return p, nil
+	if t.invalid() {
+		return nil, ErrInternal.GenByArgs("Can't find a proper physical plan for this query")
```
Comment (zhexuany) [inline plan/optimizer.go]: missing a period?

Comment (zz-jason) [reply plan/optimizer.go]: ?

**Pull request merged**